### ISSUE TITLE
Incorrect indentation for foreach after anonymous function definition

### ISSUE THREAD

**Issue by TimWolla** [squizlabs/PHP_CodeSniffer#3199]
**Describe the bug**

Incorrect indentation by `Generic.WhiteSpace.ScopeIndent.IncorrectExact` for `foreach` after anonymous function definition.

**Code sample**
```php
<?php

namespace foo;

class Foo
{
    protected function foo()
    {
        Bar::bar()
            ->bar(function (): void {
                $func = function () {
                };

                foreach ([] as $baz) { // line 14
                    ;
                }
            });
    }
}
```

**Custom ruleset**
```xml
<?xml version="1.0"?>
<ruleset>
	<file>./test/</file>
	<arg name="extensions" value="php" />
	<arg value="p"/>
	<arg name="basepath" value="."/>

	<rule ref="PSR12"/>
</ruleset>
```

**To reproduce**

1. Create a file called `test.php` with the code sample above.
2. Run `phpcs test.php`
3. See error message displayed

```
FILE: test/test.php
--------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
--------------------------------------------------------------------------
 14 | ERROR | [x] Line indented incorrectly; expected 12 spaces, found 16
 16 | ERROR | [x] Line indented incorrectly; expected 12 spaces, found 16
--------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
--------------------------------------------------------------------------
```

**Expected behavior**

I expected that error not to be reported, because I believe that the `foreach` is indented correctly. In any case phpcs should not report the `foreach` only, but the whole function body.

**Versions (please complete the following information):**

 - OS: Docker container (Debian based) on Ubuntu host.
 - PHP: 7.4.12
 - PHPCS: 3.5.8 (stable)
 - Standard: PSR12

**Additional context**

This issue might be related: #2944

**Comment by rodrigoprimo** [comment]
A similar request has been filed in the new PHP_CodeSniffer repository: PHPCSStandards/PHP_CodeSniffer#1312. Linking for visibility.
